### DISCUSSION TITLE
[Bug] Typescript - Could not find a declaration file for module

### DISCUSSION THREAD

**Discussion by fuma-nama** [cloudinary-community/next-cloudinary#129] [Q&A] [answered]
I am using `next-cloudinary` on my Next.js 13 project written in Typescript
However, I am getting an error of `Could not find a declaration file for module 'next-cloudinary'`, it seems like there's no `d.ts` file in the dist folder

I can't find `@types/next-cloudinary` in the npm registry or anything docs about using it with Typescript, does it support Typescript?

**Comment by colbyfayock** [comment, marked as answer]
Hey it's supposed to but honestly I'm a bit new to typescript and someone else implemented it for me. I'll try to look more into this.

**Reply by fuma-nama** [reply]
Thanks! I recommend using `tsup` for bundling typescript libraries and remember to define the ‘types` property in the package.json

**Reply by colbyfayock** [reply]
I created an issue for the error here: https://github.com/colbyfayock/next-cloudinary/issues/130

I also just put together a PR for fixing: https://github.com/colbyfayock/next-cloudinary/pull/131

would you be up for taking a quick look at that and seeing if anything stands out? I ended up having to ass a few `@ts-expect-error`'s because moving to tsup and getting rid of some other ignore statements threw different errors, and prioritizing getting this fixed before ironing out everything

thoughts / feedback / suggestions more than welcome

**Reply by colbyfayock** [reply]
@SonMooSans thanks for checking it out - the fix should be released in 2.3.0 - can you try it out in your project to see if we're looking good?

feel free to create a new Issue for any other Typescript related issues (or anything else)

**Reply by fuma-nama** [reply]
The type error has been fixed in 2.3.0 👍
It worked perfectly in my Typescript project
